File: node/tensortest.js
let ap = require('argparse');
let parser = new ap.ArgumentParser({
    version: '0.5.0',
    addHelp: true,
    description: 'Runs tf.matMul() on a n x n matrix'
});
parser.addArgument(
    [ '-n', '--number' ],
    {
        help: 'Optional, integer',
        type: 'int',
        defaultValue: 1024,
    }
);
parser.addArgument(
    [ '-p', '--print' ],
    {
        action: 'storeTrue',
        help: 'Optional, boolean',
        defaultValue: false,
    }
);
const args = parser.parseArgs();

let tf = require('@tensorflow/tfjs-node-gpu');

const size = args.number;

console.info("Generating a %dx%d randomUniform matrix, please standby", size, size);
let timer = process.hrtime();
let matrix = tf.initializers.randomUniform({minval: 0, maxval:1, seed: 2}).apply([size,size],'float32'); // tf.initializers.zeros().apply([size,size],"float32"); //
let d = process.hrtime(timer);
console.info("Generating a %dx%d matrix took: %ds %dms", size, size, d[0], d[1] / 1000000);

if(args.print===true){
    timer = process.hrtime();
    matrix.print();
    d = process.hrtime(timer);
    console.info("Printing a %dx%d matrix took: %ds %dms", size, size, d[0], d[1] / 1000000);
}
// Initialize cuBLAS....
let temp = tf.tensor2d([0.0,1.0,2.0,3.0],[2,2],"float32");
temp.matMul(temp,false,true);


timer = process.hrtime();
let result = matrix.matMul(matrix,false,true);
d = process.hrtime(timer);
console.info("Multiplying a %dx%d matrix took: %ds %dms", size, size, d[0], d[1] / 1000000);

File: python/tensortest.py
import tensorflow as tf
import argparse
import logging
from time import perf_counter

parser = argparse.ArgumentParser(description='Process some integers.')
parser.add_argument('--number', dest='number', action='store', type=int,
                    default=1024,
                    help='Optional, integer')
parser.add_argument('--dtype', dest='dtype', action='store', type=str,
                    default='float32',
                    choices=['float64','float32','float16','bfloat16','int64','int32','int16','int8','uint64','uint32','uint16','uint8'],
                    help='Optional, dtype')

args = parser.parse_args()

size = args.number;

print("Num GPUs Available: ", len(tf.config.experimental.list_physical_devices('GPU')))

tf.debugging.set_log_device_placement(True)

logging.info("Generating a %dx%d randomUniform matrix, please standby", size, size);
t1_start = perf_counter()
matrix = tf.random_uniform_initializer(minval=0, maxval=1, seed=2)(shape=[size,size],dtype=args.dtype);
t1_stop = perf_counter()
print("Generating a ",size,"x",size," matrix took: ", (t1_stop - t1_start), 's')

# Initialize cuBLAS....
temp = tf.constant([0.0,1.0,2.0,3.0],shape=[2,2],dtype="float32");
temp2 = tf.matmul(temp,temp,transpose_a=False,transpose_b=True);

t1_start = perf_counter()
result = tf.matmul(matrix,matrix,transpose_a=False,transpose_b=True);
t1_stop = perf_counter()
print("Multiplying a ",size,"x",size," matrix took: ", (t1_stop - t1_start), 's')